pkg/volume/csi/csi_mounter.go

```diff
@@ -289,20 +289,6 @@ func (c *csiMountMgr) TearDown() error {
 func (c *csiMountMgr) TearDownAt(dir string) error {
 	klog.V(4).Infof(log("Unmounter.TearDown(%s)", dir))
 
-	// is dir even mounted ?
-	// TODO (vladimirvivien) this check may not work for an emptyDir or local storage
-	// see https://github.com/kubernetes/kubernetes/pull/56836#discussion_r155834524
-	mounted, err := isDirMounted(c.plugin, dir)
-	if err != nil {
-		klog.Error(log("unmounter.Teardown failed while checking mount status for dir [%s]: %v", dir, err))
```

Comment:
How about rather than completely removing this check, if directory does not exist at all we do a fast return without calling CSI but if it does exist(mounted or not), we make the CSI call. This is what we do with flexvolume and it appears to have worked well in practice. cc @vladimirvivien

Reply:
I wonder if this way we won't violate a lifecycle contract that for `NodePublishVolume` we must call `NodeUnpublishVolume` (at least once) so CSI driver have a chance to release resources even in situation when the actual mount point was deleted.

Reply:
another case：https://github.com/kubernetes/kubernetes/pull/56836/files#r242694083